plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -190,8 +191,7 @@ public Stream<DeltaLakeTransactionLogEntry> getCheckpointTransactionLogEntries(
         LastCheckpoint checkpoint = lastCheckpoint.get();
         // Add entries contain statistics. When struct statistics are used the format of the Parquet file depends on the schema. It is important to use the schema at the time
         // of the Checkpoint creation, in case the schema has evolved since it was written.
-        Optional<MetadataAndProtocolEntry> metadataAndProtocol = Optional.empty();
-        if (entryTypes.contains(ADD)) {
+        if (entryTypes.contains(ADD) && metadataAndProtocol.isEmpty()) {
```

Comment:
Let's try to make it required when ADD is requested.

Reply:
This can be a follow-up

Reply:
Addressed.